authors/tests/test_signup_validators.py

```diff
@@ -0,0 +1,158 @@
+from django.test import TestCase
+from rest_framework.test import APIClient
+from rest_framework import status
+
+
+from ..apps.authentication.models import User
+
+
+class ViewTestCase(TestCase):
+    """Test suite for the api views."""
+
+    def setUp(self):
+        """Define the test client and other test variables."""
+
+        """User data with password shorter than 8"""
+        self.short_password = {
+            "user": {
+                "username": "test",
+                "email": "test@gmail.com",
+                "password": "1234"
+            }
+        }
+
+        """User data with no number in password"""
+        self.number_password = {
+            "user": {
+                "username": "test",
+                "email": "test@gmail.com",
+                "password": "abcdefgh"
+            }
+        }
+
+        """User data with no small letter in password"""
+        self.small_letter_password = {
+            "user": {
+                "username": "test",
+                "email": "test@gmail.com",
+                "password": "1234ABCD"
+            }
+        }
+
+        """User data with no capital letter in password"""
+        self.capital_letter_password = {
+            "user": {
+                "username": "test",
+                "email": "test@gmail.com",
+                "password": "1234abcd"
+            }
+        }
+
+        """User data with invalid email"""
+        self.invalid_password = {
+            "user": {
+                "username": "test",
+                "email": "testgmail.com",
+                "password": "1234abcD"
+            }
+        }
+
+        """Valid user data"""
+        self.user = self.user = User.objects.create(
+            username="newtest", email="newtest@gmail.com")
+
+        """Initialize client"""
+        self.client = APIClient()
+
+    def test_short_password(self):
+        """Test user cannot signup using password less than 8 characters"""
+        response = self.client.post(
+            '/api/users/',
+            self.short_password,
+            format='json'
+        )
+
+        self.assertIn("Ensure this field has at least 8 characters.",
+                      response.content.decode())
+        self.assertEquals(response.status_code,
+                          status.HTTP_400_BAD_REQUEST)
+
+    def test_for_small_letter(self):
+        """Test user cannot signup if password lacks small letter"""
+        response = self.client.post(
+            '/api/users/',
+            self.small_letter_password,
+            format='json'
+        )
+
+        self.assertIn("A password must contain atleast one small letter and one capital letter.",
+                      response.content.decode())
+        self.assertEquals(response.status_code,
+                          status.HTTP_400_BAD_REQUEST)
+
+    def test_for_capital_letter(self):
+        """Test user cannot signup if password lacks small letter"""
```

Comment:
Your doc strings on line 81 and 94 are the same and yet they are for different tests. Same as line 119 and 140.

Reply:
Thanks for pointing that out. I will make the appropriate changes.